### ISSUE TITLE
FilterExperimental.compute mixes concerns which makes it hard to read, also doesn't work

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#260]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/260) by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

The function is around 150 lines long. Most of the code has to do with bookkeeping what is needed to calculate each variant of filter. The code would be much clear if it had the the form

```python
def compute_filter_from_parts(Rinv_sig, Rinv_orthogonal, other):
    dostuff()
    return filter_vec, variance, etc

def compute_filt_noconst(self):
    orthogonalities = [ts(unit)]
    get_Rinv(orthogonalities)
    filt_vec, variance, etc = compute_filter_from_parts(self.Rinv_sig, 
                                                        get_Rinv(orthogonalities), other)
    return filt_vec, variance, etc

def compute_filt_noexp(self):
    dosomething(tau)
    ...
    return filt_vec, variance, etc
```

also i get an error when running `FilterExperimental.compute`

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Did the recent PR at least fix the concern that it “also doesn’t work”?

‌

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

It at least run, and I believe the orthogonal to an exponential part works.

**Comment by joefowler** [comment]
Fixed by #315, #313, #311.
